doc/driver.md

```diff
@@ -325,18 +341,43 @@ let compile_deps f =
 
 Let's now put together a list of all possible modules. We'll keep track of
 which library they're in, and whether that library is a part of `odoc` or a dependency
-library.
+library. For `odoc` libraries, we infer the implementation and interface source file
+path from the library name.
 
 ```ocaml env=e1
 let odoc_all_unit_paths = find_units ".." |> get_ok
 
+let source_files_of_odoc_module lib module_ =
+  let filename =
+    (match Astring.String.cut ~rev:true ~sep:"__" module_ with
+    | None -> module_
+    | Some (_, "") -> module_
+    | Some (_, module_) -> module_)
+    |> Stdlib.String.uncapitalize_ascii
+  in
+  match source_dir_of_odoc_lib lib with
+  | None -> None, None
+  | Some path ->
+      let path = Fpath.( / ) path filename in
+      let find_by_extension path exts =
+        List.map (fun ext -> Fpath.add_ext ext path) exts
+        |> List.find_opt (fun f -> Bos.OS.File.exists f |> get_ok)
+      in
+      let impl = find_by_extension path ["pp.ml" ;"ml-gen"; "ml"] in
```

Comment:
`.pp.ml` is for the output of a ppx, right ? If yes, we definitely want to show the original source.

If you want to handle cppo, I think the preprocessed source is placed into just `.ml` (`.cppo.ml` seems to be the original source).

Reply:
It is the output of a "string preprocessor".
It is the file that the ocaml compiler will consume and all locations will be relative to this file... (except if there are lines directives to the original file, in which case locations can maybe workish)
Here are what I see as advantages/disadvantages of using the `.pp.ml` versus using the `.ml`:
- Using the `.pp.ml`: will necessarily be of valid OCaml file, so syntax highlighting will work. Will work with Cppo and includes: we see the reconstructed file.
- Using the `.ml`: syntax highlighting will certainly fail. Will only workish if line directives are applied well. But, the user sees the same file as in the editor, which is good?

Reply:
I see. You are right, we want to take the `.pp.ml`, which is what the locations will be based on. Syntax highlighting is less of a problem compared to locations, which must be perfectly in sync.